flink-table-store-core/src/main/java/org/apache/flink/table/store/file/predicate/Like.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.predicate;
+
+import org.apache.flink.table.data.binary.BinaryStringData;
+import org.apache.flink.table.functions.SqlLikeUtils;
+import org.apache.flink.table.store.file.stats.FieldStats;
+
+import javax.annotation.Nullable;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** A {@link Predicate} to evaluate like. */
+public class Like implements Predicate {
```

Comment:
Maybe we don't need have a like now. We need a better implementation to eval `starts with`. It is similiar to `equals`.
For example:
- starts with 'abd'
- there is a file, min: 'abc' max: 'abe'
- should return true, because 'abd' is in the file.